### PR TITLE
fix: ensure filename for esm format

### LINKED ISSUE #363: Breaking change of ESM format filenames
**Bili**: 5.0.1, 5.0.4
**Node.js**: v14.0.0
**OS**: Linux (Arch)

As of v5, bili appears to be generating ESM targets (i.e. `--format esm`) as `<filename>.es.js` rather than `<filename>.esm.js`. This looks like a regression as I can't find any reference to it in the release notes or related issues.

Unless the name change is spotted and remediated, this silently breaks the `package.json` `module` entry, and — if the `files` array is used — can result in a package unwittingly being published with a missing ESM bundle.

### PR DESCRIPTION
**Pull request by egoist**
fix regression for #363, closes #363

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -457,12 +457,16 @@ export class Bundler {
...
-    const fileName = fileNameTemplate
+    let fileName = fileNameTemplate
       .replace(/\[min\]/, minPlaceholder)
       // The `[ext]` placeholder no longer makes sense
       // Since we only output to `.js` now
       // Probably remove it in the future
       .replace(/\[ext\]/, '.js')
+ 
+    if (rollupFormat === 'esm')  {
+      fileName = fileName.replace(/\[format\]/, '.esm')
```
Comment (fi3ework) [inline src/index.ts:468]:
```suggestion
      fileName = fileName.replace(/\[format\]/, 'esm')
```

File: src/index.ts [hunk 2]
```diff
@@ -457,12 +457,16 @@ export class Bundler {
...
         : getFileName
-    const fileName = fileNameTemplate
+    let fileName = fileNameTemplate
       .replace(/\[min\]/, minPlaceholder)
       // The `[ext]` placeholder no longer makes sense
       // Since we only output to `.js` now
       // Probably remove it in the future
       .replace(/\[ext\]/, '.js')
+ 
+    if (rollupFormat === 'esm')  {
```
Comment (fi3ework) [inline src/index.ts:467]:
```suggestion
    if (['es', 'esm', 'es-min', 'esm-min'].includes(rollupFormat))  {
```

Comment (egoist) [reply src/index.ts:467]: we want `es` format to continue using `.es` suffix to prevent from breaking change.

**Pull request merged**